chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -328,13 +326,25 @@
       if current_priority != attrs[:priority]
         `zypper --non-interactive modifyrepo --priority #{attrs[:priority]} #{name}`
       end
+      # check if attrs contains the :purge key with the databag name as a value
+      if attrs[:purge]
```

Comment:
Every other call to zypper (last two blocks) should probably be in a `else` of this condition.